src/database.module.ts

```diff
@@ -7,21 +7,34 @@ import { TypeOrmModule } from '@nestjs/typeorm';
     TypeOrmModule.forRootAsync({
       imports: [ConfigModule],
       inject: [ConfigService],
-      useFactory: (configService: ConfigService) => ({
-        cli: {
-          migrationDir: 'src/migrations',
-        },
-        database: configService.get<string>('DB_NAME'),
-        entities: [`${__dirname}/**/entities/*.entity.{ts,js}`],
-        host: configService.get<string>('DB_HOST'),
-        migrations: [`${__dirname}/../migrations/*.{ts,js}`],
-        migrationsRun: true,
-        password: configService.get<string>('DB_PASSWORD'),
-        port: configService.get<number>('DB_PORT'),
-        synchronize: false,
-        type: 'postgres',
-        username: configService.get<string>('DB_USER'),
-      }),
+      useFactory: (configService: ConfigService) => {
+        const isTestEnv = configService.get<string>('NODE_ENV') === 'test';
+        return {
+          cli: {
+            migrationsDir: 'src/migrations',
+          },
+          database: configService.get<string>(
+            isTestEnv ? 'TEST_DB_NAME' : 'DB_NAME',
```

Comment:
Maybe it would be cleaner to create two objects - prod_config and test_config, and return one of the configs depending on isTestEnv.